impl/src/settings.rs

```diff
@@ -483,77 +468,299 @@ fn validate_settings(
+fn parse_raze_settings(metadata: Metadata) -> Result<RazeSettings> {
+  // Workspace takes precedence
+  let workspace_level_settigns = metadata.workspace_metadata.get("raze");
+  if let Some(value) = workspace_level_settigns {
+    return parse_raze_settings_workspace(value, &metadata);
+  }
+
+  // Root packages are the next priority
+  if let Some(root_package) = metadata.root_package() {
+    if let Some(value) = root_package.metadata.get("raze") {
+      return parse_raze_settings_root_package(value, root_package);
+    }
+  }
+
+  // Attempt to load legacy settings but do not allow failures to propogate
+  if let Ok(settings) = parse_raze_settings_legacy(&metadata) {
     eprintln!(
-      "WARNING: The top-level [raze] key is deprecated. Please set [package.metadata.raze] \
-       instead."
+      "WARNING: The top-level `[raze]` key is deprecated. Please set `[workspace.metadata.raze]` or \
+       `[package.metadata.raze]` instead."
     );
-
-    return Ok(raze);
+    return Ok(settings);
   }
 
-  return Err(RazeError::Generic(
-    "Cargo.toml has no `raze`, `workspace.metadata.raze` or `package.metadata.raze` field".into(),
-  ));
+  // Finally check any package for settings
+  parse_raze_settings_any_package(&metadata)
+}
+
+struct SettingsMetadataFetcher {
+  pub cargo_bin_path: PathBuf,
 }
 
-pub fn load_settings<T: AsRef<Path>>(cargo_toml_path: T) -> Result<RazeSettings, RazeError> {
-  let path = cargo_toml_path.as_ref();
-  let mut toml = match File::open(path) {
-    Ok(handle) => handle,
-    Err(err) => {
-      return Err(RazeError::Generic(err.to_string()));
-    },
+impl MetadataFetcher for SettingsMetadataFetcher {
+  fn fetch_metadata(&self, working_dir: &Path, _include_deps: bool) -> Result<Metadata> {
+    // This fetch should not require network access.
+    MetadataCommand::new()
+      .cargo_path(&self.cargo_bin_path)
+      .no_deps()
+      .current_dir(working_dir)
+      .other_options(vec!["--offline".to_owned()])
+      .exec()
+      .with_context(|| {
+        format!(
+          "Failed to fetch Metadata with `{}` from `{}`",
+          &self.cargo_bin_path.display(),
+          working_dir.display()
+        )
+      })
+  }
+}
+
+/** Load settings used to configure the functionality of Cargo Raze */
+pub fn load_settings<T: AsRef<Path>>(
+  cargo_toml_path: T,
+  cargo_bin_path: Option<String>,
+) -> Result<RazeSettings, RazeError> {
+  // Create a MetadataFetcher from either an optional Cargo binary path
+  // or a fallback expected to be found on the system.
+  let fetcher = SettingsMetadataFetcher {
+    cargo_bin_path: cargo_bin_path
+      .unwrap_or(SYSTEM_CARGO_BIN_PATH.to_string())
+      .into(),
   };
 
-  let toml_contents = {
-    let mut contents = String::new();
-    let result = toml.read_to_string(&mut contents);
-    if let Some(err) = result.err() {
-      return Err(RazeError::Generic(err.to_string()));
+  // UNWRAP: safe due to earlier assert
```

Comment:
Which assertion / where assertion?

Reply:
I must have removed it and not noticed this. I updated the error handling there.